lib/jar.ts

```diff
@@ -138,11 +141,13 @@ async function getMavenPackageInfo(sha1: string): Promise<MavenPackageInfo> {
             ),
           );
         }
-
         if (res.response.docs.length > 1) {
-          debug('Got multiple results for sha1, only returning first one');
+          const sha1Target = path.parse(targetPath).base;
+          debug('Got multiple results for sha1, looking for', sha1Target);
+          res.response.docs.map((doc) =>
+            sha1Target.includes(doc.g) ? resolve(doc) : 0,
```

Comment:
why `map` ? you are not using the result vector at all...
I'd use lodash to find the index of the element and return it, or  `Array.filter`;